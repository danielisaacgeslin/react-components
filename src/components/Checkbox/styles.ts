import { StyleSheet } from 'aphrodite/no-important';
import { Color } from 'utils/dist/models/general';
import { hexToRgba } from 'utils/dist/utils';

import icon from '../../assets/images/check.svg';

export default StyleSheet.create({
  block: {
    appearance: 'none',
    position: 'relative',
    outline: 0,
    cursor: 'pointer',
    width: 18,
    height: 18,
    display: 'inline-block',

    ':before': {
      content: '""',
      position: 'absolute',
      display: 'block',
      width: 18,
      height: 18,
      border: `1px solid ${hexToRgba(Color.GRAY, 70)}`
    },
    ':after': {
      content: '""',
      position: 'absolute',
      display: 'block',
      width: 18,
      height: 18,
      border: `1px solid transparent`
    },
    ':checked': {
      ':before': {
        backgroundColor: Color.BLACK,
        border: `1px solid ${Color.BLACK}`
      },
      ':after': {
        backgroundPosition: 'center',
        backgroundSize: 'cover',
        backgroundImage: `url(${icon})`
      }
    },
    ':disabled': {
      opacity: 0.45
    }
  },
  checkboxType: {
    ':before': {
      borderRadius: 4
    },
    ':after': {
      borderRadius: 4
    }
  },
  radioType: {
    ':before': {
      borderRadius: '100%'
    },
    ':after': {
      borderRadius: '100%'
    }
  }
});
