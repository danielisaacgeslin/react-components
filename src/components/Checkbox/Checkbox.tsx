import * as React from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export interface ICheckboxProps {
  checked: boolean;
  disabled?: boolean;
  radio?: boolean;
}

export default class Checkbox extends React.PureComponent<ICheckboxProps> {
  public render() {
    const { checked, disabled, radio } = this.props;
    return (
      <input
        key="Checkbox"
        type="checkbox"
        className={css(styles.block, radio && styles.radioType, !radio && styles.checkboxType)}
        checked={checked}
        disabled={!!disabled}
        readOnly={true}
      />
    );
  }
}
