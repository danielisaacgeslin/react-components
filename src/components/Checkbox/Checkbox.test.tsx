import * as React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Checkbox, { ICheckboxProps } from './Checkbox';

describe('Checkbox Component', () => {
  let Component;
  let props: ICheckboxProps;

  beforeEach(() => {
    props = {
      checked: true,
      radio: false
    };
  });

  it('should render checked', () => {
    props.checked = true;
    Component = shallow(<Checkbox {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render unchecked', () => {
    props.checked = false;
    Component = shallow(<Checkbox {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render radio checked', () => {
    props.radio = true;
    props.checked = true;
    Component = shallow(<Checkbox {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render radio unchecked', () => {
    props.radio = true;
    props.checked = false;
    Component = shallow(<Checkbox {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });
});
