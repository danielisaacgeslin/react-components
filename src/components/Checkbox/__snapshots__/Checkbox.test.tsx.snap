// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component should render checked 1`] = `
.block-checkboxType {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  outline: 0px;
  cursor: pointer;
  width: 18px;
  height: 18px;
  display: inline-block;
}

.block-checkboxType:checked:before {
  background-color: #000;
  border: 1px solid #000;
}

.block-checkboxType:checked:after {
  background-position: center;
  background-size: cover;
  background-image: url(check.svg);
}

.block-checkboxType:disabled {
  opacity: 0.45;
}

.block-checkboxType:before {
  content: "";
  position: absolute;
  display: block;
  width: 18px;
  height: 18px;
  border: 1px solid rgba(213, 222, 241, 0.7);
  border-radius: 4px;
}

.block-checkboxType:after {
  content: "";
  position: absolute;
  display: block;
  width: 18px;
  height: 18px;
  border: 1px solid transparent;
  border-radius: 4px;
}

<input
  checked={true}
  className="block-checkboxType"
  disabled={false}
  readOnly={true}
  type="checkbox"
/>
`;

exports[`Checkbox Component should render radio checked 1`] = `
.block-radioType {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  outline: 0px;
  cursor: pointer;
  width: 18px;
  height: 18px;
  display: inline-block;
}

.block-radioType:checked:before {
  background-color: #000;
  border: 1px solid #000;
}

.block-radioType:checked:after {
  background-position: center;
  background-size: cover;
  background-image: url(check.svg);
}

.block-radioType:disabled {
  opacity: 0.45;
}

.block-radioType:before {
  content: "";
  position: absolute;
  display: block;
  width: 18px;
  height: 18px;
  border: 1px solid rgba(213, 222, 241, 0.7);
  border-radius: 100%;
}

.block-radioType:after {
  content: "";
  position: absolute;
  display: block;
  width: 18px;
  height: 18px;
  border: 1px solid transparent;
  border-radius: 100%;
}

<input
  checked={true}
  className="block-radioType"
  disabled={false}
  readOnly={true}
  type="checkbox"
/>
`;

exports[`Checkbox Component should render radio unchecked 1`] = `
.block-radioType {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  outline: 0px;
  cursor: pointer;
  width: 18px;
  height: 18px;
  display: inline-block;
}

.block-radioType:checked:before {
  background-color: #000;
  border: 1px solid #000;
}

.block-radioType:checked:after {
  background-position: center;
  background-size: cover;
  background-image: url(check.svg);
}

.block-radioType:disabled {
  opacity: 0.45;
}

.block-radioType:before {
  content: "";
  position: absolute;
  display: block;
  width: 18px;
  height: 18px;
  border: 1px solid rgba(213, 222, 241, 0.7);
  border-radius: 100%;
}

.block-radioType:after {
  content: "";
  position: absolute;
  display: block;
  width: 18px;
  height: 18px;
  border: 1px solid transparent;
  border-radius: 100%;
}

<input
  checked={false}
  className="block-radioType"
  disabled={false}
  readOnly={true}
  type="checkbox"
/>
`;

exports[`Checkbox Component should render unchecked 1`] = `
.block-checkboxType {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  outline: 0px;
  cursor: pointer;
  width: 18px;
  height: 18px;
  display: inline-block;
}

.block-checkboxType:checked:before {
  background-color: #000;
  border: 1px solid #000;
}

.block-checkboxType:checked:after {
  background-position: center;
  background-size: cover;
  background-image: url(check.svg);
}

.block-checkboxType:disabled {
  opacity: 0.45;
}

.block-checkboxType:before {
  content: "";
  position: absolute;
  display: block;
  width: 18px;
  height: 18px;
  border: 1px solid rgba(213, 222, 241, 0.7);
  border-radius: 4px;
}

.block-checkboxType:after {
  content: "";
  position: absolute;
  display: block;
  width: 18px;
  height: 18px;
  border: 1px solid transparent;
  border-radius: 4px;
}

<input
  checked={false}
  className="block-checkboxType"
  disabled={false}
  readOnly={true}
  type="checkbox"
/>
`;
